robot/impl/local_robot.go

```diff
@@ -439,20 +439,8 @@ func newWithResources(
 		return nil, err
 	}
 
-	// Once web service is started, start module manager and add initially
-	// specified modules.
+	// Once web service is started, start module manager
 	r.manager.startModuleManager(r.webSvc.ModuleAddress(), cfg.UntrustedEnv, logger)
-	for _, mod := range cfg.Modules {
-		// this is done in config validation but partial start rules require us to check again
-		if err := mod.Validate(""); err != nil {
-			r.logger.Errorw("module config validation error; skipping", "module", mod.Name, "error", err)
-			continue
-		}
-		if err := r.manager.moduleManager.Add(ctx, mod); err != nil {
-			r.logger.Errorw("error adding module", "module", mod.Name, "error", err)
-			continue
-		}
-	}
```

Comment:
Talked with James about these lines. He suspects they are a bug. Removing them ensures that the modules start /after/ the placeholder replacement rather than before.

I did not notice any regressions on either local or registry modules when I removed these lines. Should double/triple check though.

Reply:
Fine with it as long as we test it and it still works. Its probably just vestigal from some of the reconfigure work that was done.